lms/models/lti_role.py

```diff
@@ -0,0 +1,147 @@
+        role = role.lower()
+
+        for string in cls._V11_INSTRUCTOR_STRINGS:
+            if string in role:
+                return RoleType.INSTRUCTOR
+
+        if "learner" in role:
+            return RoleType.LEARNER
+
+        if "admin" in role:
+            return RoleType.ADMIN
+
+        return None
+
+    _V13_PREFIX = "http://purl.imsglobal.org/vocab/lis/v2"
+    _V13_ROLE_MAPPINGS = {
+        # https://www.imsglobal.org/spec/lti/v1p3/#lis-vocabulary-for-context-roles
+        # A.2.1 LIS vocabulary for system roles
+        # Core system roles
+        f"{_V13_PREFIX}/system/person#Administrator": RoleType.ADMIN,
+        f"{_V13_PREFIX}/system/person#None": RoleType.LEARNER,
+        # Non‑core system roles
+        f"{_V13_PREFIX}/system/person#AccountAdmin": RoleType.ADMIN,
+        f"{_V13_PREFIX}/system/person#Creator": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/system/person#SysAdmin": RoleType.ADMIN,
+        f"{_V13_PREFIX}/system/person#SysSupport": RoleType.ADMIN,
+        f"{_V13_PREFIX}/system/person#User": RoleType.LEARNER,
+        # A.2.2 LIS vocabulary for institution roles
+        # Core institution roles
+        f"{_V13_PREFIX}/institution/person#Administrator": RoleType.ADMIN,
+        f"{_V13_PREFIX}/institution/person#Faculty": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/institution/person#Guest": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#None": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#Other": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#Staff": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/institution/person#Student": RoleType.LEARNER,
+        # Non‑core institution roles
+        f"{_V13_PREFIX}/institution/person#Alumni": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#Instructor": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/institution/person#Learner": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#Member": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#Mentor": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/institution/person#Observer": RoleType.LEARNER,
+        f"{_V13_PREFIX}/institution/person#ProspectiveStudent": RoleType.LEARNER,
+        # A.2.3 LIS vocabulary for context roles
+        # Core context roles
+        f"{_V13_PREFIX}/membership/Administrator": RoleType.ADMIN,
+        f"{_V13_PREFIX}/membership/ContentDeveloper": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/membership/Instructor": RoleType.INSTRUCTOR,
+        # Look out for this weirdo!
+        f"{_V13_PREFIX}/membership/Learner#Instructor": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/membership/Learner": RoleType.LEARNER,
+        f"{_V13_PREFIX}/membership/Mentor": RoleType.INSTRUCTOR,
+        # Non‑core context roles
+        f"{_V13_PREFIX}/membership/Manager": RoleType.INSTRUCTOR,
+        f"{_V13_PREFIX}/membership/Member": RoleType.LEARNER,
+        f"{_V13_PREFIX}/membership/Office": RoleType.INSTRUCTOR,
```

Comment:
You could probably condense this down into something which happened to pass the tests etc, but this is very explicit, and very easy to change.